src/components/HorizontalTab/index.tsx

```diff
@@ -71,11 +66,17 @@ const HorizontalTab = ({selectedTab, onSelectedTabChange, tabs = []}: Horizontal
     <View style={styles.tabs}>
       {tabs.map((tab, index) => (
         <CustomPressable
-          testID={`horizontal-tab-${index}`}
-          key={tab.key}
+          key={index?.toString()}
           onPress={() => onSelectedTabChange(index)}
-          style={index === selectedTab ? styles.activeTabItem : styles.tabItem}>
-          <View style={styles.childTabContainer}>{tab.tabElement}</View>
+          style={
+            index === selectedTab
+              ? [
+                  styles.activeTabItem,
+                  {borderBottomColor: index === 0 ? colors.darkBlue : colors.black}
+                ]
+              : styles.tabItem
+          }>
+          <View style={styles.childTabContainer}>{tab}</View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The key for each tab item has been changed from `tab.key` to `index?.toString()`. Using indices as keys in React can cause issues with component state and performance, especially when items can be reordered, added, or removed. If the `tabs` items have unique, stable identifiers, it would be better to use those as keys. Also, the testID prop has been removed which might affect your tests.

```diff
- key={index?.toString()}
+ key={tab.key}
+ testID={`horizontal-tab-${index}`}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->